dataset/src/main/scala/frameless/functions/Udf.scala

```diff
@@ -0,0 +1,97 @@
+package frameless
+package functions
+
+import org.apache.spark.sql.catalyst.expressions.ScalaUDF
+
+/** Documentation marked "apache/spark" is thanks to apache/spark Contributors
+  * at https://github.com/apache/spark, licensed under Apache v2.0 available at
+  * http://www.apache.org/licenses/LICENSE-2.0
+  */
+trait Udf {
+
+  /** Defines a user-defined function of 1 arguments as user-defined function (UDF).
+    * The data types are automatically inferred based on the function's signature.
+    *
+    * apache/spark
+    */
+  def udf[T, A, R](f: A => R)
+  (implicit
+     reng: TypedEncoder[R]
+  ): TypedColumn[T, A] => TypedColumn[T, R] = {
+    u =>
+      implicit val aenc = u.uencoder
+
+      val scalaUdf = ScalaUDF(f, reng.targetDataType, Seq(u.expr),
```

Comment:
Sorry for line hunting, but we can save 3 lines per method and improve readability by replacing with:
```scala
  def udf[T, A, R: TypedEncocer](f: A => R): TypedColumn[T, A] => TypedColumn[T, R] = {
    u =>
      val aenc = u.uencoder

      val scalaUdf = ScalaUDF(f, TypedEncoder[R].targetDataType, Seq(u.expr),
        Seq(aenc.targetDataType))
       new TypedColumn[T, R](scalaUdf)
   }
```

AFAIK `implicit val` can be replaced with just a `val`.